react/ProductDescription.tsx

```diff
@@ -55,14 +66,31 @@ function ProductDescription(props: PropsWithChildren<Props>) {
   const { handles } = useCssHandles(CSS_HANDLES, { classes: props.classes })
   const intl = useIntl()
   const { product } = useProduct()
+  const { isMobile } = useDevice()
 
   const description = props.description ?? product?.description
 
   if (!description) {
     return null
   }
+  
+  const { collapseContent = true, title, customCollapse = false } = props
+
+  let {desktopLimitCharacters = defaultDesktopCharacters, mobileLimitCharacters = defaultMobileCharacters} = props
+  let limitCharacters = 0;
+  if(customCollapse) {
+    desktopLimitCharacters = Number.isNaN(desktopLimitCharacters) ? defaultDesktopCharacters : desktopLimitCharacters;
+    mobileLimitCharacters = Number.isNaN(mobileLimitCharacters) ? defaultMobileCharacters : mobileLimitCharacters;
+    limitCharacters = isMobile ? mobileLimitCharacters : desktopLimitCharacters;
+  }
 
-  const { collapseContent = true, title } = props
+  const [ filteredDescription, setFilteredDescription ] = useState(description.substring(0, limitCharacters));
+  const [ showMoreButton, setShowMoreButton ] = useState(true)
+
+  const changeDescription = () => {
+    setFilteredDescription(description)
+    setShowMoreButton(false)
+  }
```

Comment:
Since their behavior is always going to be related, use a single state and make them react to it. Ex:

```ts
const [isCustomCollapsed, setCustomCollapsed] = useState(customCollapse)
```

then your `changeDescription` becomes setCustomCollapsed(false)